py/picca/delta_extraction/astronomical_objects/forest.py

```diff
@@ -440,63 +361,32 @@ def rebin(self):
-            half_delta_lambda = delta_lambda / 2.
-
-            half_delta_lambda_rest_frame = (Forest.lambda_rest_frame_grid[1] -
-                                            Forest.lambda_rest_frame_grid[0]) / 2.
-            w1 = (self.lambda_ >= Forest.lambda_grid[0] - half_delta_lambda)
-            w1 = w1 & (self.lambda_ < Forest.lambda_grid[-1] + half_delta_lambda)
-            w1 = w1 & (self.lambda_ / (1. + self.z) >= Forest.lambda_rest_frame_grid[0] - half_delta_lambda_rest_frame)
-            w1 = w1 & (self.lambda_ / (1. + self.z) < Forest.lambda_rest_frame_grid[-1] + half_delta_lambda_rest_frame)
-            w1 = w1 & (self.ivar > 0.)
-            if w1.sum() == 0:
-                self.lambda_ = np.array([])
-                self.flux = np.array([])
-                self.ivar = np.array([])
-                self.transmission_correction = np.array([])
-                return [], [], [], [], []
-            self.lambda_ = self.lambda_[w1]
-            self.flux = self.flux[w1]
-            self.ivar = self.ivar[w1]
-            self.transmission_correction = self.transmission_correction[w1]
-
-            bins = find_bins(self.lambda_, Forest.lambda_grid)
-            self.lambda_ = Forest.lambda_grid[0] + bins * delta_lambda
-        else:
-            raise AstronomicalObjectError("Error in rebinning Forest. "
-                                          "Class variable 'wave_solution' "
-                                          "must be either 'lin' or 'log'. "
-                                          f"Found: {Forest.wave_solution}")
+        delta_log_lambda = Forest.log_lambda_grid[1] - Forest.log_lambda_grid[0]
+        half_delta_log_lambda = delta_log_lambda / 2.
+
+        half_delta_log_lambda_rest_frame = (Forest.log_lambda_rest_frame_grid[1] -
+                                            Forest.log_lambda_rest_frame_grid[0]) / 2.
+
+        w1 = (self.log_lambda >= Forest.log_lambda_grid[0] - half_delta_log_lambda)
+        w1 = w1 & (self.log_lambda < Forest.log_lambda_grid[-1] + half_delta_log_lambda)
+        w1 = w1 & (self.log_lambda - np.log10(1. + self.z) >=
+                   Forest.log_lambda_rest_frame_grid[0] - half_delta_log_lambda_rest_frame)
+        w1 = w1 & (self.log_lambda - np.log10(1. + self.z) <
+                   Forest.log_lambda_rest_frame_grid[-1] + half_delta_log_lambda_rest_frame)
+        w1 = w1 & (self.ivar > 0.)
+        if w1.sum() == 0:
+            self.log_lambda = np.array([])
+            self.flux = np.array([])
+            self.ivar = np.array([])
+            self.transmission_correction = np.array([])
+            return [], [], [], [], []
+        self.log_lambda = self.log_lambda[w1]
+        self.flux = self.flux[w1]
+        self.ivar = self.ivar[w1]
+        self.transmission_correction = self.transmission_correction[w1]
+
+        bins = find_bins(self.log_lambda, Forest.log_lambda_grid)
+        self.log_lambda = Forest.log_lambda_grid[0] + bins * delta_log_lambda
```

Comment:
Wait, now I'm very confused. Are we using delta_log_lambda to setup the grid, even when running with linear pixels? @iprafols

Reply:
Indeed it looks like this. I am very strongly against doing this! It will mess up the whole Pk1d computation as the number of pixels (and thus the resulting overall noise) in a wavelength bin will become strongly wavelength dependent and the resolution matrix becomes completely meaningless if we do non-linear regriddings in the analysis ever. For that purpose I'm also suggesting to merge #850 before finishing this merge to be sure we catch all the dependencies on wave solution needed for P1d

Reply:
I have now restored the rebinning function to what it should be